tests/unit/common/test_table.py

```diff
@@ -666,6 +666,147 @@ def test_repr(self):
         self.assertEqual(str(dynamic_table_region), expected)
 
 
+class DynamicTableRegionRoundTrip(H5RoundTripMixin, TestCase):
+
+    def make_tables(self):
+        self.spec2 = [
+            {'name': 'qux', 'description': 'qux column'},
+            {'name': 'quz', 'description': 'quz column'},
+        ]
+        self.data2 = [
+            ['qux_1', 'qux_2'],
+            ['quz_1', 'quz_2'],
+        ]
+
+        target_columns = [
+            VectorData(name=s['name'], description=s['description'], data=d)
+            for s, d in zip(self.spec2, self.data2)
+        ]
+        target_table = DynamicTable("target_table", 'example table to target with a DynamicTableRegion',
+                                    columns=target_columns)
+
+        self.spec1 = [
+            {'name': 'foo', 'description': 'foo column'},
+            {'name': 'bar', 'description': 'bar column'},
+            {'name': 'baz', 'description': 'baz column'},
+            {'name': 'dtr', 'description': 'DTR'},
+        ]
+        self.data1 = [
+            [1, 2, 3, 4, 5],
+            [10.0, 20.0, 30.0, 40.0, 50.0],
+            ['cat', 'dog', 'bird', 'fish', 'lizard']
+        ]
+        columns = [
+            VectorData(name=s['name'], description=s['description'], data=d)
+            for s, d in zip(self.spec1, self.data1)
+        ]
+        columns.append(DynamicTableRegion(name='dtr', description='example DynamicTableRegion',
+                                          data=[0, 1, 1, 0, 1], table=target_table))
+        table = DynamicTable("table_with_dtr", 'a test table that has a DynamicTableRegion', columns=columns)
+        return table, target_table
+
+    def setUp(self):
+        self.table, self.target_table = self.make_tables()
+        super().setUp()
+
+    def setUpContainer(self):
+        multi_container = SimpleMultiContainer('multi', [self.table, self.target_table])
+        return multi_container
+
+    def _get(self, arg):
+        mc = self.roundtripContainer()
+        table = mc.containers['table_with_dtr']
+        return table.get(arg)
+
+    def _get_nodf(self, arg):
+        mc = self.roundtripContainer()
+        table = mc.containers['table_with_dtr']
+        return table.get(arg, df=False)
+
+    def _getitem(self, arg):
+        mc = self.roundtripContainer()
+        table = mc.containers['table_with_dtr']
+        return table[arg]
+
+    def test_getitem_oor(self):
+        with self.assertRaisesRegex(IndexError, 'Row index 12 out of range'):
```

Comment:
I recommend using `self.assertRaisesWith` which tests for an exact match instead of potentially a partial match.

Reply:
I can't find any documentation on that method.

Reply:
It's actually part of `hdmf.testing.TestCase` https://hdmf.readthedocs.io/en/stable/hdmf.testing.testcase.html